### PR TITLE
[WIP] Fix type annotations for Pandas.core.indexes.datetimelike

### LINKED ISSUE #26364: Fix type annotations for pandas.core.indexes.datetimelike
Part of #25882

Current errors are:

    pandas\core\indexes\datetimelike.py:65: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:66: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:67: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:69: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:70: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:137: error: Decorated property not supported
    pandas\core\indexes\datetimelike.py:207: error: Name '_box_values' already defined on line 72
    pandas\core\indexes\datetimelike.py:761: error: Need type annotation for '_raw_methods'
    pandas\core\indexes\datetimelike.py:763: error: Need type annotation for '_raw_properties'

### PR DESCRIPTION
**Pull request by vaibhavhrt**
- [x] closes #26364 
- [ ] tests passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/datetimelike.py [hunk 1]
```diff
@@ -134,7 +135,7 @@ def values(self) -> np.ndarray:
         # Note: PeriodArray overrides this to return an ndarray of objects.
         return self._data._data
 
-    @property
+    @property  # type: ignore
```
Comment (vaibhavhrt) [inline pandas/core/indexes/datetimelike.py]: error: Decorated property not supported
Looks like people are ignoring this error for now. https://github.com/python/mypy/issues/1362

File: pandas/core/indexes/datetimelike.py [hunk 2]
```diff
@@ -758,9 +759,9 @@ class DatetimelikeDelegateMixin(PandasDelegate):
         boxed in an index, after being returned from the array
     """
     # raw_methods : dispatch methods that shouldn't be boxed in an Index
-    _raw_methods = set()
+    _raw_methods = set()  # type: Set[str]
```
Comment (vaibhavhrt) [inline pandas/core/indexes/datetimelike.py]: Not sure if its a set of `str` or `method`, need some help here.

**Comment by vaibhavhrt** [comment]
Errors that still remain are:

    pandas\core\indexes\datetimelike.py:66: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:67: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:68: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:70: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:71: error: "Callable[[Any], Any]" has no attribute "fget"
    pandas\core\indexes\datetimelike.py:208: error: Name '_box_values' already defined on line 73

Errors related to `fget` doesn't seem to have any solution yet(see https://github.com/python/mypy/issues/6185 and linked mypy issues in its comments). Maybe we should ignore it for now.

@WillAyd need some insight on why `_box_values` are declared twice and how to annotate it, or maybe ignore it too.

btw mypy tests will fail, so maybe someone with access to azure pipelines should cancel the tests.

**Comment by WillAyd** [comment]
There is already an open PR for this one so closing this as a duplicate

**Comment by vaibhavhrt** [comment]
my bad, didn't saw that

**Pull request closed without merging**